repoload/repoload.py

```diff
@@ -74,9 +85,15 @@ def get_gerrit_url():
 #   https://gerrit-review.googlesource.com/Documentation/cmd-query.html
 #   https://gerrit-review.googlesource.com/Documentation/user-search.html
 #   https://gerrit-review.googlesource.com/Documentation/json.html
-def query(url, query):
-    # TODO Make port configurable
-    cmd = "ssh -p 29418 {} gerrit query --format=JSON '{}'".format(url, query)
+def query(url: str, query: str, port: int = 29418,
+          options: Union[None, GerritOptions, Sequence[GerritOptions]] = None):
+    opt = ""
+    if isinstance(options, list):
+        opt = " ".join([o.value for o in options])
+    elif isinstance(options, GerritOptions):
+        opt = options.value
+
+    cmd = "ssh -p {port} {url} gerrit query --format=JSON {options} '{query}'".format(port=port, url=url, options=opt, query=query)
     p = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
```

Comment:
It's time to get rid of this white space splitting with `cmd.split()` which starts to break when an option or query contains a whitespace itself. That's an issue in every shell implementation and usage.

Since we have python here and `Popen` accepts a list of strings, the code should also create a list of strings instead of relying on the no whitespace in the options hope. It will break someday.

So `cmds` should be a list of strings and elements should be appended with the list functions `.append()` or `.extend()`.

In short python is programming, not shell scripting :-)